Mathlib/Analysis/InnerProductSpace/Harmonic/Constructions.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.InnerProductSpace.Harmonic.Basic
+import Mathlib.Analysis.Calculus.ContDiff.RestrictScalars
+import Mathlib.Analysis.SpecialFunctions.Complex.Analytic
+
+/-!
+# Construction of Harmonic Functions
+
+This file constructs examples of harmonic functions.
+
+If `f : ℂ → F` is complex-differentiable, then `f` is harmonic. If `F = ℂ`, then so is its real
+part, imaginary part, and complex conjugate. If `f` has no zero, then `log ‖f‖` is harmonic.
+-/
+
+open Complex InnerProductSpace Topology
+
+variable
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℂ F]
+  {f : ℂ → F} {x : ℂ}
+
+/-!
+## Harmonicity of Analytic Functions on the Complex Plane
+-/
+
+/--
+Continuously complex-differentiable functions on ℂ are harmonic.
+-/
+theorem ContDiffAt.harmonicAt (h : ContDiffAt ℂ 2 f x) : HarmonicAt f x := by
+  constructor
+  · exact h.restrict_scalars ℝ
+  · filter_upwards [h.restrictScalars_iteratedFDeriv_eventuallyEq (𝕜 := ℝ)] with a ha
+    have : (iteratedFDeriv ℂ 2 f a) (I • ![1, 1])
+        = (∏ i, I) • ((iteratedFDeriv ℂ 2 f a) ![1, 1]) :=
+      (iteratedFDeriv ℂ 2 f a).map_smul_univ (fun _ ↦ I) ![1, 1]
+    simp_all [laplacian_eq_iteratedFDeriv_complexPlane f, ← ha,
+      ContinuousMultilinearMap.coe_restrictScalars]
+
+/--
+Analytic functions on ℂ are harmonic.
+-/
+theorem AnalyticAt.harmonicAt [CompleteSpace F] (h : AnalyticAt ℂ f x) :
+    HarmonicAt f x := h.contDiffAt.harmonicAt
+
+/--
+If `f : ℂ → ℂ` is complex-analytic, then its real part is harmonic.
+-/
+theorem ContDiffAt.harmonicAt_realPart {f : ℂ → ℂ} (h : AnalyticAt ℂ f x) :
+    HarmonicAt (reCLM ∘ f) x := h.harmonicAt.comp_CLM reCLM
+
+/--
+If `f : ℂ → ℂ` is complex-analytic, then its imaginary part is harmonic.
+-/
+theorem ContDiffAt.harmonicAt_imaginaryPart {f : ℂ → ℂ} (h : AnalyticAt ℂ f x) :
+  HarmonicAt (imCLM ∘ f) x := h.harmonicAt.comp_CLM imCLM
+
+/--
+If `f : ℂ → ℂ` is complex-analytic, then its complex conjugate is harmonic.
+-/
+theorem ContDiffAt.harmonicAt_conjugate {f : ℂ → ℂ} (h : AnalyticAt ℂ f x) :
+  HarmonicAt (conjCLE ∘ f) x := (harmonicAt_comp_CLE_iff conjCLE).2 h.harmonicAt
+
+/-!
+## Harmonicity of `log ‖analytic‖`
+-/
+
+/- Helper lemma for AnalyticAt.harmonicAt_log_norm -/
+private lemma slitPlaneLemma {z : ℂ} (hz : z ≠ 0) : z ∈ slitPlane ∨ -z ∈ slitPlane := by
```

Comment:
Could this lemma be useful more broadly? Can you move this next to slitPlan, and give it a more mathlib-style name?

Reply:
Yes. Implemented as suggested.